dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -1162,3 +1190,57 @@ void CommandBarFlyoutCommandBar::ClearShadow()
         }
     }
 }
+
+bool CommandBarFlyoutCommandBar::IsExpandedUp()
+{
+    return m_isExpandedUp;
+}
+
+bool CommandBarFlyoutCommandBar::HasSecondaryOpenCloseAnimations()
+{
+    return SharedHelpers::IsAnimationsEnabled() && static_cast<bool>(m_expandedDownToCollapsedStoryboard || m_expandedUpToCollapsedStoryboard || m_collapsedToExpandedUpStoryboard || m_collapsedToExpandedDownStoryboard);
+}
+
+void CommandBarFlyoutCommandBar::AttachEventsToSecondaryStoryboards()
+{
+    winrt::IControlProtected thisAsControlProtected = *this;
+
+    const auto addDropShadowFunc = [this](auto const&, auto const&)
+    {
+        if (SharedHelpers::IsAnimationsEnabled())
+        {
+            if (auto owningFlyout = m_owningFlyout.get())
+            {
+                if (auto actualFlyout = winrt::get_self<CommandBarFlyout>(owningFlyout))
+                {
+                    actualFlyout->AddDropShadow();
+                }
+            }
+        }
+    };
+
+    m_expandedDownToCollapsedStoryboard.set(GetTemplateChildT<winrt::Storyboard>(L"ExpandedDownToCollapsed", thisAsControlProtected));
+    if (m_expandedDownToCollapsedStoryboard)
+    {
+        m_expandedDownToCollapsedStoryboardRevoker = m_expandedDownToCollapsedStoryboard.get().Completed(winrt::auto_revoke, addDropShadowFunc);
```

Comment:
>addDropShadowFunc [](start = 125, length = 17)

shouldn't we be removing the drop shadow when we go to the collapsed state?

Reply:
This should stay as add because when the overflow transitions to collapsed, it can play a shrink animation on the primary commands. The Closing event removes the shadow before the "to collapsed" animation starts and this should re-add it when the animation ends.